programs/dynamic-bonding-curve/src/instructions/creator/ix_transfer_pool_creator.rs

```diff
@@ -24,16 +27,54 @@ pub struct TransferPoolCreatorCtx<'info> {
     pub new_creator: UncheckedAccount<'info>,
 }
 
-pub fn handle_transfer_pool_creator(ctx: Context<TransferPoolCreatorCtx>) -> Result<()> {
+pub fn handle_transfer_pool_creator<'c: 'info, 'info>(
+    ctx: Context<'_, '_, 'c, 'info, TransferPoolCreatorCtx>,
+) -> Result<()> {
     let mut pool = ctx.accounts.virtual_pool.load_mut()?;
 
     let migration_progress = pool.get_migration_progress()?;
-    // avoid pool creator to do update between 2 periods
-    require!(
-        migration_progress == MigrationProgress::PreBondingCurve
-            || migration_progress == MigrationProgress::CreatedPool,
-        PoolError::NotPermitToDoThisAction
-    );
+    let config = ctx.accounts.config.load()?;
+    match migration_progress {
+        MigrationProgress::PreBondingCurve => {
+            // always work
+        }
+        MigrationProgress::CreatedPool => {
+            let migration_option = MigrationOption::try_from(config.migration_option)
+                .map_err(|_| PoolError::InvalidMigrationOption)?;
+            if migration_option == MigrationOption::MeteoraDamm {
```

Comment:
Could you add comment to indicate that once the pool is created
1. Can only transfer pool creator after LP claimed + locked
2. If the LP was claimed + locked to old creator, they have to transfer the LP to new creator + support claiming fees of locked escrow for the new creator as well